core/services/job/orm.go

```diff
@@ -252,18 +253,60 @@ func (o *orm) CreateJob(jb *Job, qopts ...pg.QOpt) error {
 					if !useForwarders && err != nil {
 						return errors.Wrapf(ErrNoSuchTransmitterKey, "%v", jb.OCR2OracleSpec.TransmitterID)
 					}
+
+					chainIDInterface, ok := jb.OCR2OracleSpec.RelayConfig["chainID"]
+					if !ok {
+						return errors.Wrap(err, "missing chainID in EVM type in OCR2 job spec")
+					}
+					newChainID, ok := chainIDInterface.(int64)
+					if !ok {
+						return errors.Wrapf(err, "failed to parse chainID in OCR2 job spec")
+					}
+
+					var specs []OCR2OracleSpec
+					err = tx.Select(&specs, `SELECT * FROM ocr2_oracle_specs WHERE relay = $1 AND contract_id = $2`,
+						"EVM", jb.OCR2OracleSpec.ContractID,
+					)
+
+					if !errors.Is(err, sql.ErrNoRows) {
+						if err != nil {
+							return errors.Wrap(err, "Unexpected error validating contract_id")
+						}
+						for _, spec := range specs {
+							chainIDValue, ok := spec.RelayConfig["ChainID"]
+							if ok {
+								chainID, ok := chainIDValue.(int64)
+								if ok {
+									if chainID == newChainID {
+										return errors.Errorf("a job with contract address %v already exists for chain ID %v", jb.OCROracleSpec.ContractAddress, chainID)
+									}
+									continue
+								}
+							}
+							return errors.Wrap(err, "Unexpected error validating contract_id")
```

Comment:
It looks like the `err` being wrapped will be nil since this happens for `!ok`. How about handling the error cases first to keep things close together and reduce indentation too:
```suggestion
							if !ok {
							...
							}
							chainID, ok := chainIDValue.(int64)
							if !ok {
							...
							}
							if chainID == newChainID {
								return errors.Errorf("a job with contract address %v already exists for chain ID %v", jb.OCROracleSpec.ContractAddress, chainID)
							}
```

Reply:
Good catch!  Should have been Errorf not Wrapf.

The order I had that way originally, but then saw a way to combine the duplicated error message.  I'll change it back though; the flow of logic is a bit easier to read the first way.